boa/src/builtins/string/mod.rs

```diff
@@ -70,8 +70,11 @@ impl String {
     #[allow(clippy::wrong_self_convention)]
     pub(crate) fn to_string(this: &mut Value, _: &[Value], _: &mut Interpreter) -> ResultValue {
         // Get String from String Object and send it back as a new value
-        let primitive_val = this.get_internal_slot("StringData");
-        Ok(Value::from(format!("{}", primitive_val)))
+        match this.get_internal_slot("StringData").data() {
+            ValueData::String(ref string) => Ok(Value::from(string.clone())),
+            // Throw expection here:
+            _ => panic!("TypeError: this is not a string"),
+        }
```

Comment:
Ideally, we should `throw` the `TypeError` here, as we do with the `RangeError` in other places. Also, this could be replaced by an `if let` and reduce indentation:
```suggestion
        if let ValueData::String(ref string) = this.get_internal_slot("StringData").data() {
            Ok(Value::from(string.clone()))
        } else {
            // TODO: Throw expection here:
            panic!("TypeError: this is not a string")
        }
```

Reply:
I my opinion using `match` in this scenario is better with the `if let` we get 80 something characters in one line (not including indent)